modules/auxiliary/scanner/smtp/smtp_enum.rb

```diff
@@ -209,8 +209,12 @@ def do_rcpt_enum(domain, usernames)
 
   def extract_words(wordfile)
     return [] unless wordfile && File.readable?(wordfile)
-    words = File.open(wordfile, "rb") {|f| f.read}
-    save_array = words.split(/\r?\n/)
-    return save_array
+
+    begin
+      File.readlines(wordfile, chomp: true)
```

Comment:
According to https://ruby-doc.org/core-2.5.3/IO.html#method-c-readlines this function also takes in a `open_args` parameter which can be passed to `open`. I guess my real question here is is this going to actually read the file as a binary file or just as a normal file? I don't see any evidence that its reading the file as a binary file here so I'm a bit cautious of this.

That being said these appear to be being used to read in wordlists which I don't think is as pressing of a concern, my only real question then would be if reading in as binary would solve encoding issues such as Chinese characters etc.

Reply:
Resolving this for now since it seems this won't really affect anything serious. If it does we can come back and revisit later.